### DISCUSSION TITLE
collapse row pivots on data grid by default

### DISCUSSION THREAD

**Discussion by f0ster** [perspective-dev/perspective#1358] [Q&A] [answered]
Is there any way to collapse row pivots on the datagrid viewer by default, on load? Thanks

**Comment by texodus** [comment, marked as answer]
[`collapse(n)`]((https://github.com/finos/perspective/tree/master/packages/perspective#viewcollapse---promise-void)) on a `<perspective-viewer>`'s `.view` property will collapse all rows to depth `n`, just like shift+click on a row.  For current perspective. you will need to ensure that
 * The `perspective-viewer-datagrid` plugin is currently loaded - calling `collapse` or `expand` on a `view` with other plugins won't currently work (e.g. D3FC).
 * The viewer has already been rendered - since this is a dataset operation, calling it before the data is fully loaded does nothing.  You can use `await viewer.load()`, or use [`flush()`]((https://github.com/finos/perspective/tree/master/packages/perspective-viewer#module_perspective-viewer..PerspectiveViewer+flush)) for this.

**Reply by f0ster** [reply]
This is very helpful, thank you 🥳

**Reply by ccarli** [reply]
hello
can this be called on a perspective widget ? 
I work in jupyterlab environment and I don't manage to group rows by default - would anyone be able to share a working example?

Thanks for the awesome library !
